app/connectors/optout/OptOutUpdateRequestModel.scala

```diff
@@ -17,30 +17,27 @@
 package connectors.optout
 
 import play.api.libs.json.{Format, Json}
-import play.mvc.Http.Status.{INTERNAL_SERVER_ERROR, NO_CONTENT}
+import play.mvc.Http.Status.NO_CONTENT
 
 object OptOutUpdateRequestModel {
 
   val optOutUpdateReason: String = "10"
 
   case class OptOutUpdateRequest(taxYear: String, updateReason: String)
-  sealed trait OptOutUpdateResponse {
-    val statusCode: Int
-  }
+  sealed trait OptOutUpdateResponse
+
   case class OptOutUpdateResponseSuccess(correlationId: String, statusCode: Int = NO_CONTENT) extends OptOutUpdateResponse
   case class ErrorItem(code: String, reason: String)
-  case class OptOutUpdateResponseFailure(correlationId: String, statusCode: Int, failures: List[ErrorItem]) extends OptOutUpdateResponse
+  case class OptOutUpdateResponseFailure(failures: List[ErrorItem]) extends OptOutUpdateResponse
 
   object OptOutUpdateResponseFailure {
-    def defaultFailure(correlationId: String = "unknown"): OptOutUpdateResponseFailure =
-      OptOutUpdateResponseFailure(correlationId,
-        INTERNAL_SERVER_ERROR,
+    def defaultFailure(): OptOutUpdateResponseFailure =
+      OptOutUpdateResponseFailure(
         List(ErrorItem("INTERNAL_SERVER_ERROR", "Request failed due to unknown error"))
       )
 
-    def notFoundFailure(correlationId: String = "unknown", url: String): OptOutUpdateResponseFailure =
-      OptOutUpdateResponseFailure(correlationId,
-        INTERNAL_SERVER_ERROR,
+    def notFoundFailure(url: String): OptOutUpdateResponseFailure =
+      OptOutUpdateResponseFailure(
```

Comment:
Would it be possible to parse the actual errors from the backend, rather than assuming we know the response? The different error responses all have different reason strings, which we could just pick out.

Reply:
actually this was a tactical solution before the BE end-point implementation and it remained after that, also 'notFoundFailure' meaning the service wasn't reachable so there would be no message to parse or pick

Reply:
done

Reply:
I think we've got there with this one now, right?